src/compiler/commandLineParser.ts

```diff
@@ -658,6 +690,8 @@ namespace ts {
         {
             name: "strictFunctionTypes",
             type: "boolean",
+            affectsSemanticDiagnostics: true,

```

Comment:
Some of these where missing affectsSemanticDiagnostics or affectsEmit and added them as well when reviewing for the new flags